src/types/models/carSales.ts

```diff
@@ -9,7 +9,7 @@ export interface Buyer {
 
 export interface CarSaleRejection {
   comment?: string
-  reason: "car-not-sold" | "wrong-buyer"
+  reason: "car-not-sold" | "wrong-buyer" | "other"
```

Comment:
I am not sure what more I need to do here. 
Nothing will change except that `comment` will become mandatory in case `reason` is "other". 
Is it something that should be handled here or in DH ?

Reply:
I'm not crazy good at this, but there is a way to define the type in a way that only when it's `other`, it has to have a `comment` and all other reasons must not have it. I think the concept is https://basarat.gitbook.io/typescript/type-system/discriminated-unions and it's something along those lines:

```
type CarSaleRejectionChoice = {
  reason: "car-not-sold" | "wrong-buyer"
};
type CarSaleRejectionOther = {
  reason: "other,
  comment: string
};
type CarSaleRejection = CarSaleRejectionChoice | CarSaleRejectionOther;
```

I'm not sure what implications it has on the app. depending how the form is returning the data, it might involve manually typing the data for each case. want to give it a try anyway?

if it causes too much of a hassle, I'd run with the type you went for. the api will validate the data too

Reply:
I think it's fine to keep the comment as optional in the type like you did, the API will return a validation error that covers the business logic in case the reason is "other" and we send an empty comment.

Reply:
Thank you Lorenzo and Christian for your feedback !